### PR TITLE
Give anatomy to creatures

### PR DESCRIPTION
**Pull request by Fris0uman**
### Summary
```SUMMARY: Infrastructure "Give anatomy to creatures"```

#### Purpose of change
Give anatomy to creatures
This is part of  #39068

#### Describe the solution
Add creature_anatomy entry to creature definition
Add related get and set functions
Moves get_all_body_parts from character/monster to creature
Moves get_random_body_parts from character/monster to creature

#### Testing
Compile without errors
Fight a monster
No apparent problem

#### Additional context
This requires  #39158 to be merged first

### CODE REVIEW

File: data/json/anatomy.json [hunk 1]
```diff
@@ -3,5 +3,10 @@
     "id": "human_anatomy",
     "type": "anatomy",
     "parts": [ "eyes", "head", "mouth", "torso", "arm_l", "arm_r", "leg_l", "leg_r", "foot_l", "foot_r", "num_bp" ]
```
Comment (FuelType-Memes) [inline data/json/anatomy.json:5]: You might be lacking hands here

Comment (Fris0uman) [reply data/json/anatomy.json:5]: right

Comment (FuelType-Memes) [reply data/json/anatomy.json:5]: left too

**Comment by mlangsdorf** [comment]
how does this work with the existing bodytype monster field?

**Comment by Fris0uman** [comment]
No interaction currently, monster keep their body made of one torso.
We could move bodytype to be an anatomy entry instead maybe.

**Comment by kevingranade** [comment]
Merged as part of #39249

**Pull request closed without merging**